src/Validator/Rules/SingleFieldSubscription.php

```diff
@@ -0,0 +1,54 @@
+<?php
+
+declare(strict_types=1);
+
+namespace GraphQL\Validator\Rules;
+
+use GraphQL\Error\Error;
+use GraphQL\Language\AST\NodeKind;
+use GraphQL\Language\AST\NodeList;
+use GraphQL\Language\AST\OperationDefinitionNode;
+use GraphQL\Language\Visitor;
+use GraphQL\Language\VisitorOperation;
+use GraphQL\Validator\ValidationContext;
+use function array_splice;
+use function count;
+use function sprintf;
+
+class SingleFieldSubscription extends ValidationRule
+{
+    public function getVisitor(ValidationContext $context)
```

Comment:
```suggestion
    /**
     * @return array<string, Closure>
     */
    public function getVisitor(ValidationContext $context): array
```

Reply:
👍 maybe even something like `* @return array<string, callable():VisitorOperation>`

Reply:
I did not know that PHPStan supports function types. Neat.

Reply:
phpcs does not like this one though...

```
FILE: src/Validator/Rules/SingleFieldSubscription.php
----------------------------------------------------------------------
FOUND 1 ERROR AFFECTING 1 LINE
----------------------------------------------------------------------
 21 | ERROR | @return annotation of method
    |       | \GraphQL\Validator\Rules\SingleFieldSubscription::getVisitor()
    |       | does not specify type hint for items of its traversable
    |       | return value.
    |       | (SlevomatCodingStandard.TypeHints.TypeHintDeclaration.MissingTraversableReturnTypeHintSpecification)
----------------------------------------------------------------------
```

Just this it does like (and same goes for phpstan):

```php
    /**
     * @return array<string, callable>
     */
```

Reply:
I think that's good enough.

Reply:
ah right, that should be fixed in https://github.com/webonyx/graphql-php/pull/562 nvm